### ISSUE TITLE
Android startActivityForResult

### ISSUE THREAD

**Issue by mobilemindtec** [NativeScript/NativeScript#1835]
Hi,

I have a problem when I call startActivityForResult. The app navigation no more  work. My code:
**
Google Plus Login**

`    var signInIntent = com.google.android.gms.auth.api.Auth.GoogleSignInApi.getSignInIntent(mGoogleApiClient);
     var previousResult = _AndroidApplication.onActivityResult;

      _AndroidApplication.onActivityResult = function (requestCode, resultCode, data) {
     
        _AndroidApplication.onActivityResult = previousResult;
     
        if (requestCode === RC_SIGN_IN && resultCode === android.app.Activity.RESULT_OK) {
            var result = com.google.android.gms.auth.api.Auth.GoogleSignInApi.getSignInResultFromIntent(data);
            handleSignInResult(result);
        }else{
            mFailCallback()
        }
     }

    var act = _AndroidApplication.foregroundActivity || _AndroidApplication.startActivity;
    act.startActivityForResult(signInIntent, RC_SIGN_IN); `

Google Login view  is showed. I cancel the login or confirm the login. So I go to page register or page home:

**Register view call**

`  var topmost = frameModule.topmost()
   topmost.transition = { name: "fade" };
   topmost.navigate({
     moduleName: "views/register/register",
     animated: true,    
  });`

But the navigation not more work after call startActivityForResult. This is a bug? I'm using the last version, 1.7.1

**Steps:**
1 - Call the third-party app (google plus or facebook)
2 - Confirme or cancel interation
3 - Navegate to new page - not work

**Comment by tsonevn** [comment]
Hi @mobilemindtec,
We found that similar issue exists. This problem is scheduled for fixing.
We will do everything possible to solve the problem.

Duplicate of #1820

**Comment by PanayotCankov** [comment]
@mobilemindtec Can you try to apply the fix in this PR: #1839 by hand in your `node_modules/tns-core-modules` and confirm if the fix will solve the problem at your side?

**Comment by mobilemindtec** [comment]
Hi, 

To test I changed my application.android.js and add in onActivityResumed method:

`androidApp.foregroundActivity = activity;`

Now work fine! But I get a new bug in my context.. this is my code:
`  user.register()
    .then(function() {
      dialogsModule
        .alert("Sua conta foi criada com sucesso!")
        .then(function() {
          login.successLogin()
        });
    }).catch(function(error) {
      dialogsModule
        .alert({
          message: error,
          okButtonText: "OK"
        });
    })`

user.register() return fetch that connect with my server and authenticate.. but the alert dialog never is showed. This is the log in logcat:

`invalid stream - problem with parameter iterator in /data/user/0/com.myapp/code_cache/secondary-dexes/com.tns.Async_CompleteCallback_ftns_modules_http_http-request_l28_c24__-1458845290249-1.jar for method void com.tns.gen.com.tns.Async_CompleteCallback_ftns_modules_http_http-request_l28_c24__.onComplete(java.lang.Object, java.lang.Object)`

To work, I changed the dialogs.android.js module  in method createAlertDialog. I change:

`var alert = new android.app.AlertDialog.Builder(appmodule.android.currentContext);`

to:

`    var ctx = _AndroidApplication.foregroundActivity || _AndroidApplication.startActivity;
    var alert = new android.app.AlertDialog.Builder(ctx);
`

And now alert work fine too.. this is a new bug with a fix.
